plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt

```diff
@@ -55,23 +55,19 @@ class StmtConversionVisitor : FirVisitor<Exp, StmtConversionContext>() {
         val expr = returnExpression.result.accept(this, data)
         // TODO: respect return-based control flow
         val returnVar = data.signature.returnVar
-        data.addStatement(Stmt.LocalVarAssign(returnVar.toLocalVar(), expr))
+        data.addStatement(Stmt.LocalVarAssign(returnVar.toLocalVar(), expr.withType(returnVar.type.type)))
```

Comment:
The double `.type` looks confusing.  Can we use `.viperType` as a shorthand?  (Perhaps a `Typed` interface that provides both properties, with a default implementation for one?)

Reply:
I think adding a `.viperType` shorthand is the most practical.